File: Protogen/loxim_protocol/cpp/protocol/packages/V_sc_sendvaluesPackage.cpp
#include <string.h>
#include <stdlib.h>
#include "V_sc_sendvaluesPackage.h"
#include "ptools/PackageBufferWriter.h"
#include "ptools/PackageBufferReader.h"

using namespace protocol;

V_sc_sendvaluesPackage::V_sc_sendvaluesPackage(\
		int64_t a_rootValueId,\
		int64_t a_oBundlesCount,\
		int64_t a_oBidCount,\
		int64_t a_pVidCount
		,bool a_bnull_oBundlesCount
		,bool a_bnull_oBidCount
		,bool a_bnull_pVidCount
		)
{
	rootValueId=a_rootValueId;
	oBundlesCount=a_oBundlesCount;
	bnull_oBundlesCount=a_bnull_oBundlesCount;
	oBidCount=a_oBidCount;
	bnull_oBidCount=a_bnull_oBidCount;
	pVidCount=a_pVidCount;
	bnull_pVidCount=a_bnull_pVidCount;
}

V_sc_sendvaluesPackage::V_sc_sendvaluesPackage()
{
	rootValueId=0;
	oBundlesCount=0;
	bnull_oBundlesCount=false;
	oBidCount=0;
	bnull_oBidCount=false;
	pVidCount=0;
	bnull_pVidCount=false;
}

V_sc_sendvaluesPackage::~V_sc_sendvaluesPackage()
{
}

void V_sc_sendvaluesPackage::serializeW(PackageBufferWriter *writer)
{
	Package::serializeW(writer);
	writer->writeVaruint(rootValueId);
	if (bnull_oBundlesCount) writer->writeVaruintNull(); else writer->writeVaruint(oBundlesCount);
	if (bnull_oBidCount) writer->writeVaruintNull(); else writer->writeVaruint(oBidCount);
	if (bnull_pVidCount) writer->writeVaruintNull(); else writer->writeVaruint(pVidCount);
}

bool V_sc_sendvaluesPackage::deserializeR(PackageBufferReader *reader)
{
	if (!Package::deserializeR(reader)) return false;
	{bool b; rootValueId=reader->readVaruint(b);}
	oBundlesCount=reader->readVaruint(bnull_oBundlesCount);
	oBidCount=reader->readVaruint(bnull_oBidCount);
	pVidCount=reader->readVaruint(bnull_pVidCount);
	return true;
}

bool V_sc_sendvaluesPackage::equals(Package *_package)
{
	if (!_package) return false;
	if (_package->getPackageType()!=this->getPackageType()) return false;
	if (!Package::equals(_package)) return false;
	V_sc_sendvaluesPackage* p=(V_sc_sendvaluesPackage*)_package;
	if (!(getRootValueId()==p->getRootValueId())) return false;
	if (getBnull_oBundlesCount()!=p->getBnull_oBundlesCount()) return false;
	if ((!bnull_oBundlesCount)&&(!(getOBundlesCount()==p->getOBundlesCount()))) return false;
	if (getBnull_oBidCount()!=p->getBnull_oBidCount()) return false;
	if ((!bnull_oBidCount)&&(!(getOBidCount()==p->getOBidCount()))) return false;
	if (getBnull_pVidCount()!=p->getBnull_pVidCount()) return false;
	if ((!bnull_pVidCount)&&(!(getPVidCount()==p->getPVidCount()))) return false;
	return true;
}

File: Protogen/loxim_protocol/java/src/main/java/pl/edu/mimuw/loxim/protocol/packages/V_sc_sendvaluesPackage.java
package pl.edu.mimuw.loxim.protocol.packages;

import java.util.EnumSet;
import pl.edu.mimuw.loxim.protogen.lang.java.template.ptools.Package;
import pl.edu.mimuw.loxim.protogen.lang.java.template.exception.ProtocolException;
import pl.edu.mimuw.loxim.protogen.lang.java.template.streams.PackageInputStreamReader;
import pl.edu.mimuw.loxim.protogen.lang.java.template.streams.PackageOutputStreamWriter;


import pl.edu.mimuw.loxim.protocol.enums.*;


public  class V_sc_sendvaluesPackage extends Package {
		public final static short ID=(short)32;
	
    	/**
**/
    	private Long rootValueId=new Long(0);
    	/**
**/
    	private Long oBundlesCount=new Long(0);
    	/**
**/
    	private Long oBidCount=new Long(0);
    	/**
**/
    	private Long pVidCount=new Long(0);
    	
	public V_sc_sendvaluesPackage()
	{
	}

		public V_sc_sendvaluesPackage(
		    	Long a_rootValueId,
	        	Long a_oBundlesCount,
	        	Long a_oBidCount,
	        	Long a_pVidCount    	)
	{
        	rootValueId=a_rootValueId;
        	oBundlesCount=a_oBundlesCount;
        	oBidCount=a_oBidCount;
        	pVidCount=a_pVidCount;
    
	}
		
		@Override
	public long getPackageType() {
		return (long)ID;
	}
		
	@Override
	protected void deserializeW(PackageInputStreamReader reader)
			throws ProtocolException {
		super.deserializeW(reader);
		
				try{
	    			rootValueId=reader.readVaruint();
					oBundlesCount=reader.readVaruint();
					oBidCount=reader.readVaruint();
					pVidCount=reader.readVaruint();
				}catch(Exception e)
		{
			throw new ProtocolException(e);
		}
			}
	
	@Override
	protected void serializeW(PackageOutputStreamWriter writer)
			throws ProtocolException {
		super.serializeW(writer);
				try{
	    			writer.writeVaruint(rootValueId);		
					writer.writeVaruint(oBundlesCount);		
					writer.writeVaruint(oBidCount);		
					writer.writeVaruint(pVidCount);		
				}catch(Exception e)
		{
			throw new ProtocolException(e);
		}	
			}
	
	@Override
	public boolean equals(Object obj) {
		if(!super.equals(obj))
			return false;
		V_sc_sendvaluesPackage p=(V_sc_sendvaluesPackage)obj;
	    if (!(((rootValueId==p.getRootValueId())||((rootValueId!=null)&&(p.getRootValueId()!=null)&&((rootValueId).equals(p.getRootValueId())))))) 
return false;
		if (!(((oBundlesCount==p.getOBundlesCount())||((oBundlesCount!=null)&&(p.getOBundlesCount()!=null)&&((oBundlesCount).equals(p.getOBundlesCount())))))) 
return false;
		if (!(((oBidCount==p.getOBidCount())||((oBidCount!=null)&&(p.getOBidCount()!=null)&&((oBidCount).equals(p.getOBidCount())))))) 
return false;
		if (!(((pVidCount==p.getPVidCount())||((pVidCount!=null)&&(p.getPVidCount()!=null)&&((pVidCount).equals(p.getPVidCount())))))) 
return false;
				return true;
	}
	
//================= GETTERS AND SETTERS ===================

        	public Long getRootValueId(){return rootValueId;};
		
    	public void setRootValueId(Long a_rootValueId){ rootValueId=a_rootValueId;};
		
        	public Long getOBundlesCount(){return oBundlesCount;};
		
    	public void setOBundlesCount(Long a_oBundlesCount){ oBundlesCount=a_oBundlesCount;};
		
        	public Long getOBidCount(){return oBidCount;};
		
    	public void setOBidCount(Long a_oBidCount){ oBidCount=a_oBidCount;};
		
        	public Long getPVidCount(){return pVidCount;};
		
    	public void setPVidCount(Long a_pVidCount){ pVidCount=a_pVidCount;};
		
    
	 
}


